src/Input.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { useState } from "react"
+import { Input as AntdInput, Button } from "antd"
+
+type Category = {
+    name: string
+}
+type InputProps = {
+    placeholder: string
+    onSubmit: (value: string) => void
+}
+const Input = ({ placeholder, onSubmit }: InputProps) => {
+
+    const [value, setValue] = useState<string>("")
+
+    const handleButtonClick = (event: React.MouseEvent<HTMLElement, MouseEvent>) => {
+        onSubmit(value)
+        setValue("")
+    }
+    const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
Ini harusnya gak perlu dibuat component baru dulu. Terlalu dini untuk bikin component baru, karena component barunya gak begitu compleks

Selengkapnya baca disini : 
https://kentcdodds.com/blog/prop-drilling
https://kentcdodds.com/blog/when-to-break-up-a-component-into-multiple-components/